addons/mail/static/src/core/common/attachment_model.js

```diff
@@ -41,6 +44,28 @@ export class Attachment extends FileModelMixin(Record) {
         }
         return `${this.create_date.monthLong}, ${this.create_date.year}`;
     }
+
+    /** Remove the given attachment globally. */
+    delete() {
+        if (this.tmpUrl) {
+            URL.revokeObjectURL(this.tmpUrl);
+        }
+        super.delete();
+    }
+
+    /**
+     * Delete the given attachment on the server as well as removing it
+     * globally.
+     */
+    async remove() {
+        this.delete();
```

Comment:
move delete after (ct fwport in progress)

Reply:
Also probably `remove` could be turned into side-effect of `delete()`